docs/APP-SNAPSHOT.md

```diff
@@ -0,0 +1,1053 @@
+
+##### Reporting Rollback Error
+
+If an error occurs during the rollback process, the device reports the error by
+setting the `ErrorInfo` field in the `ZInfoSnapshot` message of the
+corresponding snapshot (the snapshots field in the `ZInfoApp` message will still
+contain a`ZInfoSnapshot` message for the snapshot used in the rollback)
+
+### Snapshot Deletion API
+
+The snapshot deletion process allows the controller to request the deletion of a
+specific snapshot on the device. The device then attempts to delete the
+specified snapshot and reports the outcome of the deletion to the controller.
+
+#### Controller Triggering Snapshot Deletion
+
+The controller triggers the deletion of a snapshot by sending
+an `AppInstanceConfig` message to the device with the `snapshots` field updated
+to exclude the snapshot that needs to be deleted. The `snapshots` field is a
+list of `SnapshotDesc` messages, each containing information about a snapshot.
+The controller should remove the `SnapshotDesc` message corresponding to the
+snapshot that needs to be deleted from the snapshots field.
+
+#### Device Reporting Deletion Result
+
+After receiving the snapshot deletion request, the device attempts to delete the
+specified snapshot. The device then reports the outcome of the deletion in
+the `ZInfoApp` message, which includes the `snapshots` – a list
+of `ZInfoSnapshot` messages, each containing information about a snapshot.
+
+##### Reporting Deletion Success
+
+The device indicates a successful deletion by removing the `ZInfoSnapshot`
+message corresponding to the deleted snapshot from the snapshots field in
+the `ZInfoApp` message. The absence of the `ZInfoSnapshot` message for the
+specified snapshot ID confirms that the snapshot has been successfully deleted.
+
+##### Reporting Deletion Error
+
+Due to the current implementation limitation, the device always reports
+successful snapshot deletion and does not report any errors that may occur
+during the snapshot deletion process. Nevertheless, the expected logic for error
+reporting would be as described below.
+
+If an error occurs during the snapshot deletion process, the device reports the
+error by setting the `ErrorInfo field in the ZInfoSnapshot message of the
+corresponding snapshot. The snapshots field in the ZInfoApp message will contain
+a ZInfoSnapshot message for the snapshot that was attempted to be deleted. The
+id field in the ZInfoSnapshot message will match the ID specified in the
+deletion request, and the snap_err field will contain the error information.
+
+##### Important Limitation
+
+It is important to note that in the current implementation, EVE reports
+successful snapshot deletion immediately after receiving the request without
+waiting for the corresponding cleanup actions to be completed on the EVE side.
+As a result, EVE currently cannot report any errors that may occur during the
+snapshot deletion process.
+
+## EVE Components
```

Comment:
Nit: In the docs we use "component" as a general term (electroic component) and pieces of the build (grub might be referred to as a component), but for the different EVE microservices like zedagent we tend to use either "agent" or "microservice". Thus I think it makes sense to follow that naming scheme (but referring to the solution having a config and volume components earlier in the document is perfectly ok.)